### ISSUE TITLE
passing a symbol for "format" silently disables validation

### ISSUE THREAD

**Issue by Roguelazer** [davishmcclurg/json_schemer#197]
```ruby
JSONSchemer.schema({'format' => 'date'}).valid?("foo")
JSONSchemer.schema({'format' => :date}).valid?("foo")
```

This is very surprising! I feel like `fetch_format` might want to cast the input to a string. What do you think?

**Comment by davishmcclurg** [comment]
I agree it's annoying. `format` isn't a special case, though, so changing this would mean supporting symbols for all schema values. I reluctantly added support for symbol schema keys (ie, `{format: 'date'}` works), but keys are more constrained than values. I'm still open to supporting symbol values, though likely not soon.

In the mean time, if you have schemas that may not be valid, you can use `JSONSchemer.valid_schema?` to check them:

```
>> JSONSchemer.valid_schema?({'format' => 'date'})
=> true
>> JSONSchemer.valid_schema?({'format' => :date})
=> false
```
